src/modules/flow/json/json.c

```diff
@@ -186,28 +273,163 @@ send_token_packet(struct sol_flow_node *node, struct sol_json_scanner *scanner,
         SOL_STR_SLICE_PRINT(slice));
 }
 
-
 static int
 json_object_key_process(struct sol_flow_node *node, struct sol_json_object_key *mdata)
 {
+    struct sol_json_token value, token;
     struct sol_json_scanner scanner;
-    struct sol_json_token value;
 
     if (!mdata->key[0] || !mdata->json_object)
         return 0;
 
     sol_json_scanner_init(&scanner, mdata->json_object->mem,
         mdata->json_object->size);
+    if (_sol_json_loop_helper_init(&scanner, &token,
+        SOL_JSON_TYPE_OBJECT_START) != SOL_JSON_LOOP_REASON_OK)
+        return false;
 
-    if (json_object_search_for_token(&scanner, mdata->key, &value))
+    if (json_object_search_for_token(&scanner,
+        sol_str_slice_from_str(mdata->key), &value))
         return send_token_packet(node, &scanner, mdata->json_object, &value);
 
     return sol_flow_send_error_packet(node, EINVAL,
         "JSON object doesn't contain key %s", mdata->key);
+}
+
+static int
+_get_brackets_index(const struct sol_str_slice slice, struct sol_str_slice *current, bool *allocated)
+{
+    const char *end, *p, *p2;
+    char *dest = NULL, *last;
+
+    end = slice.data + slice.len;
+    for (p = slice.data + 2; p < end; p = p2 + 1) {
+        p2 = memchr(p, '\'', end - p);
+        if (!p2 || p2 == end - 2)
+            break;
+        if (!dest) {
+            dest = malloc(slice.len - 5);
```

Comment:
Too many magic numbers and variables with meaningless names. I'm not sure what they mean.